raft.go

```diff
@@ -1229,7 +1229,7 @@ func stepLeader(r *raft, m pb.Message) error {
 
 				if refused != "" {
 					r.logger.Infof("%x ignoring conf change %v at config %s: %s", r.id, cc, r.prs.Config, refused)
-					m.Entries[i] = pb.Entry{Type: pb.EntryNormal}
+					return ErrProposalDropped
```

Comment:
I don't know the side effect of dropping other entries in the same `m.Entries` for loop. 

How about moving it til the end of stepLeader function, especially after `bcastAppend` so other entries handling are not impacted just to be safe?

Reply:
the loop is only processing config changes, see `if cc != nil {`. So, I think it's ok to drop here. Also, my understanding is that we have to drop full proposal, there is no per entry `ErrProposalDropped`.

Reply:
I see, thanks for the explanation!